### ISSUE TITLE
Badges are copied when pasting a resource from cart

### ISSUE THREAD

**Issue by nicolasjacquemet** [donhinkelman/moodle-block_sharing_cart#127]
We're experiencing a bug with the plugin when used in a course with badges. When you put a resource in your cart and then paste it, all the badges from the course will be copied to the target course.
Steps to reproduce : 
- Create à course
- Create a simple badge in it
- Add cart block
- Create a resource (a book for example)
- Copy the resource to sharing cart
- Copy the resource to a course using "Copy to course" of the block.
- Every badges from the source course will be copied to the target course. It also happens if the source and destination course is the same, all the badges will be duplicated then.

Plugin : version 3.11, release 2 2022080202
Moodle : version 3.11.9 build 20220822

**Comment by donhinkelman** [comment]
Thank you Nicolas for reporting this bug and listing the steps to reproduce it. I will assign this to Praxis.

**Comment by frederikmillingpytlick** [comment]
Hi @nicolasjacquemet,

We are aware of the issue and as it is an issue in Moodle Core, we have created a ticket for Moodle. 
See: https://tracker.moodle.org/browse/MDL-43697

It seems that our patch has been merged into Moodle 4.0 and above, so upgrading should fix this behaviour. 

If you would not like to upgrade to Moodle 4.0 you could try to see if the Moodle devs would merge it into Moodle 3.11 as well.

**Comment by nicolasjacquemet** [comment]
Thanks for your answers, but are we sure this is the same issue?
In my case it has nothing to do with deleted badges or course backup restoration.
In doubt, I tested it on on a Moodle 4.0.3 and the issue is still present.
I copy-pasted a resource with the sharing cart plugin in a course with a badge and my test badge was duplicated : 

![image](https://user-images.githubusercontent.com/3415139/188631485-59cf7a32-b331-4bc2-bce3-2dd7d9c4bf80.png)

**Comment by frederikmillingpytlick** [comment]
Hi @nicolasjacquemet,

We don't currently have the time to deep dive into the sharing cart at the moment, but will take a look when we get the time.

If you are looking to speed up the fixing of this issue, you are more than welcome to submit a patch for us to look at.

**Comment by marxjohnson** [comment]
I have followed the instructions to reproduce this on MOODLE_400_STABLE with the master branch of block_sharing_cart and was unable to do so. I backed up and restored a File resource from a course with a badge to a course with no badges. The destination course received the resource, but no badges. I have attached the .mbz file created by Sharing Cart, which does not contain the badge.
[Sharingcart-small files-1668172879.0094.mbz.zip](https://github.com/donhinkelman/moodle-block_sharing_cart/files/9990515/Sharingcart-small.files-1668172879.0094.mbz.zip)
